src/FSharp.Markdown/HtmlFormatting.fs

```diff
@@ -182,11 +182,18 @@ let rec formatParagraph (ctx:FormattingContext) paragraph =
       ctx.Writer.Write("</p>")
   | HorizontalRule(_) ->
       ctx.Writer.Write("<hr />")
+  | CodeBlock(String.StartsWithWrapped ("[", "]") (ParseCommands cmds, String.TrimStart code)) 
```

Comment:
Currently, code formatting is not handled in FSharp.Markdown but only in FSharp.Literate (which adds formatting for F# snippets based on compiler service and also formatting for code snippets). This is done by adding `LiterateParagraph` type ([here](https://github.com/tpetricek/FSharp.Formatting/blob/123131fbf8a00eff4fcf7d3f07a58a972758eca2/src/FSharp.Literate/Document.fs#L27)) and adding special formatting for it ([here](https://github.com/tpetricek/FSharp.Formatting/blob/123131fbf8a00eff4fcf7d3f07a58a972758eca2/src/FSharp.Literate/Transformations.fs#L319)).

Given this, I wonder what is the right structure for adding Github flavored markdown.

In a perfect world, I think the ideal solution would be:

 * Support [fenced code blocks](http://spec.commonmark.org/0.15/#code-fence) as specified by CommonMark in FSharp.Markdown
 * Add support for GitHub-style language tags in FSharp.Literate (so that we can reuse formatting that is already there for other languages and also nice formatting for F#)

Do you think it would make sense to move the language-specific stuff in your PR from FSharp.Formatting to FSharp.Literate? (If that sounds like too much work, we can start with what you have and do the rest later...)